Python2.7 compatibility

NotSqrt opened the pull request:
* fix the few code incompatibilities with six
* add tox to test locally with various python versions

Fix for #23 

tox is callable with :
```sh
LDAP_AUTH_URL=ldap://localhost LDAP_AUTH_SEARCH_BASE=dc=example,dc=com
LDAP_AUTH_TEST_USER_USERNAME=test_user
LDAP_AUTH_TEST_USER_PASSWORD=test_password tox
```

etianen commented:
Thanks!

The Travis build shows a few issues with Python 3.2. Could you take a look
(I'm on holiday ATM).
On Tue, 22 Sep 2015 at 11:19, NotSqrt <notifications@github.com> wrote:

>
>    - fix the few code incompatibilities with six
>    - add tox to test locally with various python versions
>
> Fix for #23 <https://github.com/etianen/django-python3-ldap/issues/23>
>
> tox is callable with :
>
> LDAP_AUTH_URL=ldap://localhost LDAP_AUTH_SEARCH_BASE=dc=example,dc=com
> LDAP_AUTH_TEST_USER_USERNAME=test_user
> LDAP_AUTH_TEST_USER_PASSWORD=test_password tox
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/etianen/django-python3-ldap/pull/24
> Commit Summary
>
>    - Python2.7 compatibility
>
> File Changes
>
>    - *M* .gitignore
>    <https://github.com/etianen/django-python3-ldap/pull/24/files#diff-0>
>    (1)
>    - *M* .travis.yml
>    <https://github.com/etianen/django-python3-ldap/pull/24/files#diff-1>
>    (3)
>    - *M* django_python3_ldap/conf.py
>    <https://github.com/etianen/django-python3-ldap/pull/24/files#diff-2>
>    (4)
>    - *M* django_python3_ldap/tests.py
>    <https://github.com/etianen/django-python3-ldap/pull/24/files#diff-3>
>    (12)
>    - *M* django_python3_ldap/utils.py
>    <https://github.com/etianen/django-python3-ldap/pull/24/files#diff-4>
>    (3)
>    - *M* setup.py
>    <https://github.com/etianen/django-python3-ldap/pull/24/files#diff-5>
>    (4)
>    - *A* tox.ini
>    <https://github.com/etianen/django-python3-ldap/pull/24/files#diff-6>
>    (27)
>
> Patch Links:
>
>    - https://github.com/etianen/django-python3-ldap/pull/24.patch
>    - https://github.com/etianen/django-python3-ldap/pull/24.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/etianen/django-python3-ldap/pull/24>.
>

NotSqrt commented:
The new version of coverage.py (4.0) has dropped support for python3.2 ..

etianen commented:
Annoying!

I'll investigate and merge this next week on Monday, when I'm back!
On Tue, 22 Sep 2015 at 11:29, NotSqrt <notifications@github.com> wrote:

> The new version of coverage.py (4.0) has dropped support for python3.2 ..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/etianen/django-python3-ldap/pull/24#issuecomment-142241339>
> .
>

NotSqrt commented:
Oh, I fixed the issue, it's just the latest release from 2 days ago that dropped support for python3.2 (since it's no longer maintained), so I added "coverage<4.0" as requirement in travis.

etianen commented:
Thanks! I can't look at this until Monday, but I don't anticipate any
problems with merging it in then!

Thanks for looking into this. I was holding off using this on my old Python
2 sites, so you just made my life easier! :D
On Tue, 22 Sep 2015 at 14:54, NotSqrt <notifications@github.com> wrote:

> Oh, I fixed the issue, it's just the latest release from 2 days ago that
> dropped support for python3.2 (since it's no longer maintained), so I added
> "coverage<4.0" as requirement in travis.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/etianen/django-python3-ldap/pull/24#issuecomment-142295925>
> .
>

praseodym commented:
Dajgno 1.9 drops support for Python 3.2 and 3.3, so dropping support for these versions might make sense. +1 for 2.7 support though!

NotSqrt commented:
Django 1.8 is an LTS, and will be supported until at least April 2018, so you might want to keep Python 3.2 and 3.3 for some time ;)

etianen commented:
This looks excellent.

The only thing I'd say is that we don't need six as a dependency. Django bundles a copy of six, so we should probably use that.

NotSqrt commented:
Fixed !

etianen commented:
This is great! Merging!
